torch/distributed/_shard/sharded_tensor/__init__.py

```diff
@@ -3,31 +3,24 @@
 import copy
 import functools
 import torch
-from torch.distributed._shard.sharding_spec import (
-    ChunkShardingSpec,
-    ShardingSpec,
-)
-from torch.distributed._shard.sharding_spec._internals import (
-    get_chunked_dim_size,
-    get_split_size,
-)
+import torch.distributed._shard.sharding_spec as shard_spec
 from typing import List
 
 from .api import (
     _register_sharded_op,
     Shard,
-    ShardMetadata,
     ShardedTensor,
     ShardedTensorMetadata,
     TensorProperties,
     _PartialTensor,
 )
+from .metadata import ShardMetadata  # noqa: F401
 from .utils import load_with_process_group
 import torch.distributed as dist
 from torch.distributed import distributed_c10d
 
 
-def empty(sharding_spec: ShardingSpec,
+def empty(sharding_spec: shard_spec.ShardingSpec,
```

Comment:
This is purely bc of recursive dependency, we use `ShardedTensor` in `sharding_spec` package, so we can't initialize `sharding_spec` here, otherwise it will trigger recursive imports.